### ISSUE TITLE
Crash during simple plan

### ISSUE THREAD

**Issue by henrydobson** [hashicorp/terraform#15951]
### Terraform Configuration Files
*gpg-zip*
https://www.dropbox.com/s/zg3xdpdukry3x7w/tf_crash_hcl.gpg?dl=0

### Crash Output
https://www.dropbox.com/s/9cgkcx0568bf0vo/crash.log.gpg?dl=0

### Expected Behavior
terraform plan should produce plan.

### Actual Behavior
terraform crashed.

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform workspace new utility`
3. `terraform plan`

### Important Factoids
Crash occurred after a small number of terraform imports.

**Comment by vancluever** [comment]
Hey @henrydobson - sorry that you have hit this!

Was wondering - is this for a brand new config (no state), or for a config with an existing state? If you could let us know that would be great!

Thanks!

**Comment by henrydobson** [comment]
Brand new then I imported some resources. So I guess not really brand new. 

I found some errors that I couldn't correct in the state file just now (couldn't remove resources) so I'm going to delete it and start again. Will post back with the result.

**Comment by henrydobson** [comment]
Yes, that seems to have done the trick. Plan is working correctly.

The original state file seemed had some sg rules as groups - that's probably my error - 

This issue can be closed I think. I suspect the crash was caused by incorrectly renaming state resources to the wrong type (aws_security_group and aws_security_group_rule in this case). Perhaps that would be nice safety feature for the future.

**Comment by vancluever** [comment]
Thanks @henrydobson! Good to know.

That's definitely an interesting case. The point of crash seemed to be a part of the scale-out refresh logic, so basically where TF thinks that you are trying to increase the count of a resource that already exists in state. The [line of the crash](https://github.com/hashicorp/terraform/blob/61971f2ddeb6912c65107cefaa6113ae4edb2545/terraform/node_resource_refresh.go#L218) is a line that references a specific node's configuration, which lines up with your account of the naming drift in the state (I'm guessing it thought it had a specific resource, but of course configuration did not technically exist for it).

I'm going to leave this open for now just to track any possible safety feature we can do in the future to try and prevent this from happening - but I'm glad you worked it out! :+1:
